### PR TITLE
New stable API version 2022-12-12 for CORS deviceprovisioningservices microsoft.devices 

### PR DESCRIPTION
**Pull request by fmjosuel**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  *Dec 16, 2022*
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  *Dec 09, 2022*
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 1]
```diff
@@ -0,0 +1,2571 @@
+{
```
Comment (visingla-ms) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1]: There is one more PR, with a different new API version than this: [https://github.com/Azure/azure-rest-api-specs/pull/21736](https://github.com/Azure/azure-rest-api-specs/pull/21736). That also similar changes, which one to review?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 2]
```diff
@@ -0,0 +1,2571 @@
...
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule[AccessRightsDescription]"
+          }
+        },
+        "enableDataResidency": {
+          "type": "boolean",
+          "description": "Optional.\nIndicates if the DPS instance has Data Residency enabled, removing the cross geo-pair disaster recovery."
+        },
+        "portalOperationsHostName": {
+          "type": "string",
+          "description": "Portal endpoint to enable CORS for this provisioning service."
```
Comment (visingla-ms) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1921]: >Portal endpoint to enable CORS for this provisioning service. [](http://example.com/codeflow?start=26&length=61)

Just curious, if you are looking for Azure Portal endpoint here, using which the client made this call? FYI: standard client headers which are passed to RPs by ARM, including "referrer" header: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#proxy-request-header-modifications](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#proxy-request-header-modifications).

Please remove ARMChangesRequest label, after providing the answer to this question, unless you have some further changes to make.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 3]
```diff
@@ -0,0 +1,2571 @@
...
+      }
+    },
+    "ErrorDetails": {
+      "description": "Error details.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code.",
+          "format": "int32",
+          "type": "integer",
```
Comment (visingla-ms) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2306]: These are highlighted as Breaking Changes. Please take a look, and get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com>.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 4]
```diff
@@ -0,0 +1,2571 @@
...
+          "description": "The resource location."
+        },
+        "resourcegroup": {
+          "type": "string",
+          "description": "The resource group of the resource."
+        },
+        "subscriptionid": {
+          "type": "string",
+          "description": "The subscription id of the resource."
+        },
```
Comment (visingla-ms) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: Why are these required? These are always there in Operation Resource URI, not able to derive from there?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 5]
```diff
@@ -0,0 +1,2571 @@
...
+            "operation": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Name of the operation."
+            }
+          }
+        }
+      }
+    },
+    "ErrorDetails": {
```
Comment (visingla-ms) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2299]: >ErrorDetails [](http://example.com/codeflow?start=5&length=12)

You can/should use common-types instead of defining these locally here, as those will be standard and compliant to RPC. See: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/types.json).

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 1 above)
Comment (fmjosuel) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1]: Yes, please ignore the #21736 one (closed). 

Also for context. The changes in this API version have already been approved in a private preview version. But we were not able to publish the changes to the public repo. As a result we were asked (by OpenAPI support) to create another PR manually in public repo. 

Ask: can we leverage that previous approval to expedite review on this pull request? We had gone through breaking change review as well. 

Please advise. 

https://github.com/Azure/azure-rest-api-specs-pr/pull/8870

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 2 above)
Comment (fmjosuel) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1921]: Not sure I fully understand the question, but by Portal, I meant an internal team within Azure IoT that will consume the endpoint that we are adding.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 3 above)
Comment (fmjosuel) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2306]: This is part of the pre-approved changes in private preview (per comment above). Let me know if we need to still go through another breaking change review.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 4 above)
Comment (Selekena) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: The changes made here beside the addition of the "portalOperationsHostName" were all motivated by S360 alerts raised against our service. The alerts wanted our traffic to match the swagger backing it.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 1 above)
Comment (fmjosuel) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1]: Any update on the above? @visingla-ms 
We would like to move along with this since it is a dependency for two features launch for us. Thank you!

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 2 above)
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1921]: >description [](http://example.com/codeflow?start=11&length=11)

+readonly

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 4 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: We should remove them, and stop returning them, in this new API version, since they're not actually allowed according to the RPC.

Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: Please also refactor by not declaring resource, and inheriting "allOf": TrackedResource or ProxyResource instead from common types.

Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: Resource definitions - use TrackedResource or ProxyResource in commontypes
Instead of defining your own resource class with required top level properties, 'id', 'type', 'name', and 'location' and 'tags' (for TrackedResource) its recommended to inherit from ProxyResource or TrackedResource in common-types v2 or v3.json. (The v2 definition does NOT include systemData, and is otherwise similar to the v3 one).
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56)

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 3 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2306]: +1 on using common types to conform to RPC standard, instead of declaring your own error details

Error responses - use common types
There are definitions of error responses and details in commontypes types.json that you can use instead of declaring your own.
See e.g. ErrorResponse
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L300](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L300)
And ErrorDetail
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#LL258C21-L258C21](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#LL258C21-L258C21)

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 6]
```diff
@@ -0,0 +1,2571 @@
...
+      "type": "string"
+    },
+    "resourceGroupName": {
+      "name": "resourceGroupName",
+      "description": "The name of the resource group that contains the provisioning service.",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2545]: Refactoring.

Common parameters best practice - use common types
There are definitions of common parameters in common-types/v3/types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter,
See
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630)

This should be non-breaking if you are following the normal pattern, but may look breaking if your definitions of the parameters were somehow nonstandard. In your case if you're going from preview -> GA then I think you should preferably take that breaking change now.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 7]
```diff
@@ -0,0 +1,2571 @@
...
+    "groupId": {
+      "name": "groupId",
+      "in": "path",
+      "description": "The name of the private link resource",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "privateEndpointConnectionName": {
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2562]: >privateEndpointConnectionName [](http://example.com/codeflow?start=5&length=29)

There's a standard definition of this parameter also somewhere, let me find it...

Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2562]: [https://github.com/Azure/azure-rest-api-specs/blob/3e27c70e7c02c07b458bc0e94716c3d82d3fdd19/specification/common-types/resource-management/v3/privatelinks.json#LL173C39-L173C39](https://github.com/Azure/azure-rest-api-specs/blob/3e27c70e7c02c07b458bc0e94716c3d82d3fdd19/specification/common-types/resource-management/v3/privatelinks.json#LL173C39-L173C39)

or

[https://github.com/Azure/azure-rest-api-specs/blob/3e27c70e7c02c07b458bc0e94716c3d82d3fdd19/specification/common-types/resource-management/v4/privatelinks.json#LL181C9-L181C9](https://github.com/Azure/azure-rest-api-specs/blob/3e27c70e7c02c07b458bc0e94716c3d82d3fdd19/specification/common-types/resource-management/v4/privatelinks.json#LL181C9-L181C9)

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 8]
```diff
@@ -0,0 +1,2571 @@
...
+      "x-ms-parameter-location": "method"
+    },
+    "groupId": {
+      "name": "groupId",
+      "in": "path",
+      "description": "The name of the private link resource",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2561]: We should put this in privatelinks.json too, huh.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 9]
```diff
@@ -0,0 +1,2571 @@
...
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2470]: Refactoring:

IF this is a resource type... THEN you should inherit allOf ProxyResource from common-types here also. This should be nonbreaking.

BUT... seems like you only actually return this from a POST operation?
In that case it isn't actually a resource, its just a HTTP response, so its not required to have 'name', 'id', 'type' properties like resources are when you GET them.

Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2470]: I would suggest to remove all of the properties here that you are returning which are not actually used.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 10]
```diff
@@ -0,0 +1,2571 @@
...
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2469]: > "The resource type." [](http://example.com/codeflow?start=24&length=21)

Of which resource?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 11]
```diff
@@ -0,0 +1,2571 @@
...
+        }
+      }
+    },
+    "VerificationCodeResponse": {
+      "type": "object",
+      "description": "Description of the response of the verification code.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of certificate.",
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2453]: >Name of certificate.", [](http://example.com/codeflow?start=26&length=22)

Of which resource?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 12]
```diff
@@ -0,0 +1,2571 @@
...
+          "readOnly": true
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Request etag."
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2459]: >"Request etag." [](http://example.com/codeflow?start=25&length=15)

Requests don't have etags. RESOURCES have etags.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 13]
```diff
@@ -0,0 +1,2571 @@
...
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2465]: Of which resource?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 9 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2470]: You also don't need the 'properties' wrapper for POST responses.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 14]
```diff
@@ -0,0 +1,2571 @@
...
+      "required": [
+        "sku",
+        "properties"
+      ]
+    },
+    "IotDpsPropertiesDescription": {
+      "description": "the service specific properties of a provisioning service, including keys, linked iot hubs, current state, and system generated properties such as hostname and idScope",
+      "type": "object",
+      "properties": {
+        "state": {
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1821]: >state [](http://example.com/codeflow?start=9&length=5)

+readonly

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 15]
```diff
@@ -0,0 +1,2571 @@
...
+          }
+        },
+        "privateEndpointConnections": {
+          "description": "Private endpoint connections created on this IotHub",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "provisioningState": {
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1869]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

+readonly

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 16]
```diff
@@ -0,0 +1,2571 @@
...
+          "description": "The ARM provisioning state of the provisioning service.",
+          "type": "string"
+        },
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1872]: You are recommended to model provisioningState using an enum.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 2 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:1921]: Isn't the referer header passed to RPs by ARM is... an ARM address?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 4 above)
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2236]: these are nonstandard casings.

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 17]
```diff
@@ -0,0 +1,2571 @@
...
+        }
+      }
+    },
+    "ErrorDetails": {
+      "description": "Error details.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code.",
+          "format": "int32",
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2305]: >int32 [](http://example.com/codeflow?start=21&length=5)

This was flagged as breaking, but is a S360 fix also... is it?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 13 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2465]: Is a verification code response a resource?

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json (hunk 3 above)
Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2306]: If you're going to make breaking changes to the Error Response ANYWAY, wouldn't it make more sense to just be refactoring to using the standard definitions?? If they match your service behavior, that is.

Comment (TimLovellSmith) [reply specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2306]: Turns out your service is nonstandard for returning int. ;-(

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json [hunk 18]
```diff
@@ -0,0 +1,2571 @@
...
+    },
+    "resourceName": {
+      "name": "resourceName",
+      "in": "path",
+      "description": "The name of the provisioning service.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "groupId": {
```
Comment (TimLovellSmith) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json:2554]: >groupId [](http://example.com/codeflow?start=5&length=7)

I don't see most other services defining a groupId parameter for use in private links apis, so I wonder what is different. I think they would normally call the parameter you are calling 'groupIds' 'privateLinkName' or something like that.
But its breaking, so never mind for now.

**Review by TimLovellSmith — APPROVED** [review]
:shipit:

**Comment by TimLovellSmith** [comment]
As you can see I have a bunch of feedback, but its not really ARM review blockers per se. You might want to discuss when is the right time to fix RPC violations with breaking changes reviewers -- since it will possibly be needed to fix certain bugs you find out about later.

**Comment by fmjosuel** [comment]
Private preview PR link: https://github.com/Azure/azure-rest-api-specs-pr/pull/8870

**Comment by fmjosuel** [comment]
Commenting here for context to everyone - These changes have already been cleared for breaking change review in private preview (see [merged PR here](https://github.com/Azure/azure-rest-api-specs-pr/pull/8870)). But we had an issue promoting the changes to a public GA API and were suggested to start the GA api version manually (thus this current PR).

**Comment by fmjosuel** [comment]
> Commenting here for context to everyone - These changes have already been cleared for breaking change review in private preview (see [merged PR here](https://github.com/Azure/azure-rest-api-specs-pr/pull/8870)). But we had an issue promoting the changes to a public GA API and were suggested to start the GA api version manually (thus this current PR).

Specific reference to the breaking change review: https://github.com/Azure/azure-rest-api-specs-pr/pull/8519#issuecomment-1251842254
